### PR TITLE
Fix create revision and improve view invisibility

### PR DESCRIPTION
**Pull request by grindtildeath**
Related to :
- [ ] https://github.com/OCA/sale-workflow/pull/396

### CODE REVIEW

File: sale_order_revision/models/sale_order.py [hunk 1]
```diff
@@ -19,13 +28,13 @@ class SaleOrder(models.Model):
...
     revision_number = fields.Integer('Revision',
-                                     copy=False)
+                                     copy=False,
+                                     default=0)
     unrevisioned_name = fields.Char('Order Reference',
                                     copy=True,
                                     readonly=True)
-    active = fields.Boolean('Active',
-                            default=True,
-                            copy=True)
```
Comment (JayVora-SerpentCS) [inline sale_order_revision/models/sale_order.py]: any reasons for removing active field?

Comment (grindtildeath) [reply sale_order_revision/models/sale_order.py]: My mistake, just fixed it !

**Comment by grindtildeath** [comment]
@JayVora-SerpentCS 
Hi 
I did a few changes to improve the general behaviour as I need this module in a project and it wasn't working well after my previous update. 
Now I did a few changes so we have a broader coverage on the tests and it doesn't fail with odoo standard copy().
Please check it out and improve the PR on OCA.

**Comment by JayVora-SerpentCS** [comment]
Dear @grindtildeath , Would you please update the PR? I see conflicts here.

**Comment by grindtildeath** [comment]
Resolved by : https://github.com/JayVora-SerpentCS/sale-workflow/pull/2

**Pull request closed without merging**